[Automation] java run fails for io.quarkus:quarkus-bootstrap-app-model:3.34.0

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.34.0" ./gradlew clean javaTest -Pcoordinates="io.quarkus:quarkus-bootstrap-app-model:3.33.1"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25270673198/job/74092419682
Last 300 lines of the log:
```console
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: io.quarkus:quarkus-bootstrap-app-model:3.33.1
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED

DirectoryPathTreeTest > serializesAndDeserializesDirectoryTreeState() FAILED
    org.opentest4j.AssertionFailedError: expected: io.quarkus.paths.DirectoryPathTree@295c1b3f</tmp/junit10840214510654295533> but was: io.quarkus.paths.DirectoryPathTree@3b26fcda</tmp/junit10840214510654295533>
        at app//org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
        at app//org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
        at app//org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
        at app//org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1142)
        at app//io_quarkus.quarkus_bootstrap_app_model.DirectoryPathTreeTest.serializesAndDeserializesDirectoryTreeState(DirectoryPathTreeTest.java:48)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

5 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 8s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 6 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.quarkus:quarkus-bootstrap-app-model:3.33.1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 17s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 3.34.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.34.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.34.0" ./gradlew clean test -Pcoordinates="io.quarkus:quarkus-bootstrap-app-model:3.33.1"]
========================================================================================
 io.quarkus:quarkus-bootstrap-app-model:3.33.1:3.34.0 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 985ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
========================================================================================
 io.quarkus:quarkus-bootstrap-app-model:3.33.1:3.34.0 stage java run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 889ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources NO-SOURCE
> Task :testClasses

> Task :test FAILED

DirectoryPathTreeTest > serializesAndDeserializesDirectoryTreeState() FAILED

    org.opentest4j.AssertionFailedError: expected: io.quarkus.paths.DirectoryPathTree@7e74c053</tmp/junit4110568504268347091> but was: io.quarkus.paths.DirectoryPathTree@3f770c42</tmp/junit4110568504268347091>
5 tests completed, 1 failed
        at app//org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)

        at app//org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
        at app//org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
FAILURE: Build failed with an exception.
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
        at app//org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1142)
        at app//io_quarkus.quarkus_bootstrap_app_model.DirectoryPathTreeTest.serializesAndDeserializesDirectoryTreeState(DirectoryPathTreeTest.java:48)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'java run' for 3.34.0 failed with exit code 1.
FAILED[java run][3.34.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.34.0" ./gradlew clean javaTest -Pcoordinates="io.quarkus:quarkus-bootstrap-app-model:3.33.1"]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.34.0" ./gradlew clean javaTest -Pcoordinates="io.quarkus:quarkus-bootstrap-app-model:3.33.1"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25270673198/job/74092419577
Last 300 lines of the log:
```console
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: io.quarkus:quarkus-bootstrap-app-model:3.33.1
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED


5 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 8s
DirectoryPathTreeTest > serializesAndDeserializesDirectoryTreeState() FAILED
    org.opentest4j.AssertionFailedError: expected: io.quarkus.paths.DirectoryPathTree@7068bcd</tmp/junit7811500336491679264> but was: io.quarkus.paths.DirectoryPathTree@66163393</tmp/junit7811500336491679264>
        at app//org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
        at app//org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
        at app//org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
        at app//org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1142)
        at app//io_quarkus.quarkus_bootstrap_app_model.DirectoryPathTreeTest.serializesAndDeserializesDirectoryTreeState(DirectoryPathTreeTest.java:48)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 6 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.quarkus:quarkus-bootstrap-app-model:3.33.1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 18s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 3.34.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.34.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.34.0" ./gradlew clean test -Pcoordinates="io.quarkus:quarkus-bootstrap-app-model:3.33.1"]
========================================================================================
 io.quarkus:quarkus-bootstrap-app-model:3.33.1:3.34.0 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 961ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
========================================================================================
 io.quarkus:quarkus-bootstrap-app-model:3.33.1:3.34.0 stage java run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 941ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources NO-SOURCE
> Task :testClasses


> Task :test FAILED

DirectoryPathTreeTest > serializesAndDeserializesDirectoryTreeState() FAILED
    org.opentest4j.AssertionFailedError: expected: io.quarkus.paths.DirectoryPathTree@1b08a43f</tmp/junit3367917911906573266> but was: io.quarkus.paths.DirectoryPathTree@a12a3c7</tmp/junit3367917911906573266>
5 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

        at app//org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
        at app//org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
        at app//org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
BUILD FAILED in 2s
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
        at app//org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1142)
        at app//io_quarkus.quarkus_bootstrap_app_model.DirectoryPathTreeTest.serializesAndDeserializesDirectoryTreeState(DirectoryPathTreeTest.java:48)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'java run' for 3.34.0 failed with exit code 1.
FAILED[java run][3.34.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.34.0" ./gradlew clean javaTest -Pcoordinates="io.quarkus:quarkus-bootstrap-app-model:3.33.1"]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.34.0" ./gradlew clean javaTest -Pcoordinates="io.quarkus:quarkus-bootstrap-app-model:3.33.1"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25270673198/job/74092419669
Last 300 lines of the log:
```console
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: io.quarkus:quarkus-bootstrap-app-model:3.33.1
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED

DirectoryPathTreeTest > serializesAndDeserializesDirectoryTreeState() FAILED
    org.opentest4j.AssertionFailedError: expected: io.quarkus.paths.DirectoryPathTree@295c1b3f</tmp/junit7759991731865816101> but was: io.quarkus.paths.DirectoryPathTree@3b26fcda</tmp/junit7759991731865816101>
        at app//org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
        at app//org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
        at app//org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
        at app//org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1142)

5 tests completed, 1 failed        at app//io_quarkus.quarkus_bootstrap_app_model.DirectoryPathTreeTest.serializesAndDeserializesDirectoryTreeState(DirectoryPathTreeTest.java:48)



FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 7s
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 6 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.quarkus:quarkus-bootstrap-app-model:3.33.1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 16s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 3.34.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.34.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.34.0" ./gradlew clean test -Pcoordinates="io.quarkus:quarkus-bootstrap-app-model:3.33.1"]
========================================================================================
 io.quarkus:quarkus-bootstrap-app-model:3.33.1:3.34.0 stage javac compile
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 921ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
========================================================================================
 io.quarkus:quarkus-bootstrap-app-model:3.33.1:3.34.0 stage java run
========================================================================================
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 911ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources NO-SOURCE
> Task :testClasses


> Task :test FAILED
5 tests completed, 1 failed


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s
DirectoryPathTreeTest > serializesAndDeserializesDirectoryTreeState() FAILED
    org.opentest4j.AssertionFailedError: expected: io.quarkus.paths.DirectoryPathTree@295c1b3f</tmp/junit2089516125581199810> but was: io.quarkus.paths.DirectoryPathTree@3b26fcda</tmp/junit2089516125581199810>
        at app//org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
        at app//org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
        at app//org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
        at app//org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1142)
        at app//io_quarkus.quarkus_bootstrap_app_model.DirectoryPathTreeTest.serializesAndDeserializesDirectoryTreeState(DirectoryPathTreeTest.java:48)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.33.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'java run' for 3.34.0 failed with exit code 1.
FAILED[java run][3.34.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.34.0" ./gradlew clean javaTest -Pcoordinates="io.quarkus:quarkus-bootstrap-app-model:3.33.1"]
```

--- a/metadata/io.quarkus/quarkus-bootstrap-app-model/3.34.0/reachability-metadata.json
+++ b/metadata/io.quarkus/quarkus-bootstrap-app-model/3.34.0/reachability-metadata.json
@@ -1,0 +1,194 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.model.ApplicationModelBuilder"
+      },
+      "type": "ch.qos.logback.classic.Logger"
+    },
+    {
+      "type": "io.quarkus.bootstrap.workspace.LazySourceDir",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.util.BootstrapUtils"
+      },
+      "type": "io.quarkus.bootstrap.model.DefaultApplicationModel",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.util.BootstrapUtils"
+      },
+      "type": "io.quarkus.maven.dependency.ArtifactDependency",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.util.BootstrapUtils"
+      },
+      "type": "io.quarkus.maven.dependency.GACTV",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.util.BootstrapUtils"
+      },
+      "type": "io.quarkus.maven.dependency.ResolvedArtifactDependency",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.workspace.LazySourceDir"
+      },
+      "type": "io.quarkus.paths.PathFilter",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.paths.DirectoryPathTree"
+      },
+      "type": "io.quarkus.paths.PathFilter",
+      "serializable": true
+    },
+    {
+      "type": "io.quarkus.paths.DirectoryPathTree",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.util.BootstrapUtils"
+      },
+      "type": "io.quarkus.paths.PathList",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.paths.PathTreeWithManifest"
+      },
+      "type": "java.lang.Runtime",
+      "methods": [
+        {
+          "name": "version",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.paths.PathTreeWithManifest"
+      },
+      "type": "java.lang.Runtime$Version",
+      "methods": [
+        {
+          "name": "version",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.util.BootstrapUtils"
+      },
+      "type": "java.lang.String",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.util.BootstrapUtils"
+      },
+      "type": "java.util.ArrayList",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.workspace.LazySourceDir"
+      },
+      "type": "java.util.ArrayList",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.paths.DirectoryPathTree"
+      },
+      "type": "java.util.ArrayList",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.util.BootstrapUtils"
+      },
+      "type": "java.util.CollSer",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.workspace.LazySourceDir"
+      },
+      "type": "java.util.HashMap",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.util.BootstrapUtils"
+      },
+      "type": "java.util.ImmutableCollections$ListN",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.util.BootstrapUtils"
+      },
+      "type": "java.util.ImmutableCollections$MapN",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.util.BootstrapUtils"
+      },
+      "type": "java.util.ImmutableCollections$SetN",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.workspace.LazySourceDir"
+      },
+      "type": "java.util.regex.Pattern",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.paths.DirectoryPathTree"
+      },
+      "type": "java.util.regex.Pattern",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.model.ApplicationModelBuilder"
+      },
+      "type": "org.apache.log4j.LogManager"
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.model.ApplicationModelBuilder"
+      },
+      "type": "org.apache.logging.log4j.Logger"
+    },
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.model.ApplicationModelBuilder"
+      },
+      "type": "org.jboss.logmanager.LogManager"
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "io.quarkus.bootstrap.model.ApplicationModelBuilder"
+      },
+      "glob": "META-INF/services/org.jboss.logging.LoggerProvider"
+    }
+  ]
+}

--- a/metadata/io.quarkus/quarkus-bootstrap-app-model/index.json
+++ b/metadata/io.quarkus/quarkus-bootstrap-app-model/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "3.34.0",
+    "tested-versions" : [
+      "3.34.0"
+    ],
+    "allowed-packages" : [
+      "io.quarkus"
+    ]
+  },
+  {
     "metadata-version" : "3.33.1",
     "source-code-url" : "https://repo.maven.apache.org/maven2/io/quarkus/quarkus-bootstrap-app-model/$version$/quarkus-bootstrap-app-model-$version$-sources.jar",
     "repository-url" : "https://github.com/quarkusio/quarkus",

--- a/metadata/io.quarkus/quarkus-bootstrap-app-model/index.json
+++ b/metadata/io.quarkus/quarkus-bootstrap-app-model/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "3.34.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/io/quarkus/quarkus-bootstrap-app-model/$version$/quarkus-bootstrap-app-model-$version$-sources.jar",
+    "repository-url" : "https://github.com/quarkusio/quarkus",
+    "test-code-url" : "https://github.com/quarkusio/quarkus/tree/$version$/independent-projects/bootstrap/app-model/src/test",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/io/quarkus/quarkus-bootstrap-app-model/$version$/quarkus-bootstrap-app-model-$version$-javadoc.jar",
+    "description" : "Quarkus Bootstrap App Model provides Java classes for representing the dependency graph, workspace modules, platform imports, capabilities, JVM options, and other metadata used while Quarkus bootstraps and builds an application. It is part of Quarkus' bootstrap infrastructure and is used by resolvers, build tools, and extension tooling to exchange a structured model of an application and its artifacts.",
     "tested-versions" : [
       "3.34.0"
     ],

--- a/stats/io.quarkus/quarkus-bootstrap-app-model/3.34.0/execution-metrics.json
+++ b/stats/io.quarkus/quarkus-bootstrap-app-model/3.34.0/execution-metrics.json
@@ -1,0 +1,99 @@
+{
+  "fix_java_run_fail:2026-05-05": {
+    "timestamp": "2026-05-05T21:18:31.713024Z",
+    "library": "io.quarkus:quarkus-bootstrap-app-model:3.34.0",
+    "starting_commit": "e33c6648e3bf21bce3b5fc86bedca12082ee0a77",
+    "ending_commit": "2caeb019174a73fc6dbfff9de39464bf9b6395f7",
+    "previous_library": "io.quarkus:quarkus-bootstrap-app-model:3.33.1",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "3.34.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 15,
+            "totalCalls": 15
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 15,
+        "totalCalls": 15
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 1785,
+          "missed": 15283,
+          "ratio": 0.104582,
+          "total": 17068
+        },
+        "line": {
+          "covered": 427,
+          "missed": 3380,
+          "ratio": 0.112162,
+          "total": 3807
+        },
+        "method": {
+          "covered": 123,
+          "missed": 884,
+          "ratio": 0.122145,
+          "total": 1007
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "3.33.1",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 15,
+            "totalCalls": 15
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 15,
+        "totalCalls": 15
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 1910,
+          "missed": 15067,
+          "ratio": 0.112505,
+          "total": 16977
+        },
+        "line": {
+          "covered": 447,
+          "missed": 3351,
+          "ratio": 0.117694,
+          "total": 3798
+        },
+        "method": {
+          "covered": 128,
+          "missed": 871,
+          "ratio": 0.128128,
+          "total": 999
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 23639,
+      "cached_input_tokens_used": 203264,
+      "output_tokens_used": 2341,
+      "iterations": 1,
+      "cost_usd": 0.1718,
+      "tested_library_loc": 427,
+      "metadata_entries": 49,
+      "previous_library_metadata_entries": 49,
+      "code_coverage_percent": 11.22,
+      "previous_library_coverage_percent": 11.77
+    },
+    "artifacts": {
+      "test_file": "tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/src/test/java/io_quarkus/quarkus_bootstrap_app_model/LazySourceDirTest.java",
+      "metadata_file": "metadata/io.quarkus/quarkus-bootstrap-app-model/3.34.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/io.quarkus/quarkus-bootstrap-app-model/3.34.0/stats.json
+++ b/stats/io.quarkus/quarkus-bootstrap-app-model/3.34.0/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "3.34.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 15,
+          "totalCalls" : 15
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 15,
+      "totalCalls" : 15
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 1785,
+        "missed" : 15283,
+        "ratio" : 0.104582,
+        "total" : 17068
+      },
+      "line" : {
+        "covered" : 427,
+        "missed" : 3380,
+        "ratio" : 0.112162,
+        "total" : 3807
+      },
+      "method" : {
+        "covered" : 123,
+        "missed" : 884,
+        "ratio" : 0.122145,
+        "total" : 1007
+      }
+    }
+  } ]
+}

--- a/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/.gitignore
+++ b/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/build.gradle
+++ b/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "io.quarkus:quarkus-bootstrap-app-model:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/gradle.properties
+++ b/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = io.quarkus:quarkus-bootstrap-app-model:3.34.0
+library.version = 3.34.0
+metadata.dir = io.quarkus/quarkus-bootstrap-app-model/3.34.0/

--- a/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/settings.gradle
+++ b/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'io.quarkus.quarkus-bootstrap-app-model_tests'

--- a/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/src/test/java/io_quarkus/quarkus_bootstrap_app_model/BootstrapUtilsTest.java
+++ b/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/src/test/java/io_quarkus/quarkus_bootstrap_app_model/BootstrapUtilsTest.java
@@ -1,0 +1,78 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package io_quarkus.quarkus_bootstrap_app_model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import io.quarkus.bootstrap.model.ApplicationModel;
+import io.quarkus.bootstrap.model.ApplicationModelBuilder;
+import io.quarkus.bootstrap.util.BootstrapUtils;
+import io.quarkus.maven.dependency.ArtifactCoords;
+import io.quarkus.maven.dependency.ResolvedDependencyBuilder;
+import io.quarkus.paths.PathList;
+
+public class BootstrapUtilsTest {
+    private static final String APP_COORDINATES = "org.example:bootstrap-utils-test-app::jar:test-version";
+
+    @TempDir
+    Path tempDir;
+
+    @Test
+    @SuppressWarnings("removal")
+    public void serializesAndDeserializesQuarkusModel() throws Exception {
+        final ApplicationModel model = newApplicationModel();
+        final Path modelFile = tempDir.resolve("plain").resolve("app-model.dat");
+
+        BootstrapUtils.serializeAppModel(model, modelFile);
+
+        assertTrue(Files.exists(modelFile));
+
+        final ApplicationModel copy = BootstrapUtils.deserializeQuarkusModel(modelFile);
+
+        assertSameApplicationArtifact(model, copy);
+        assertFalse(Files.exists(modelFile));
+    }
+
+    @Test
+    @SuppressWarnings("removal")
+    public void writesAndReadsAppModelWithWorkspaceId() throws Exception {
+        final ApplicationModel model = newApplicationModel();
+        final Path modelFile = tempDir.resolve("workspace").resolve("app-model.dat");
+        final int workspaceId = 42;
+
+        BootstrapUtils.writeAppModelWithWorkspaceId(model, workspaceId, modelFile);
+
+        assertTrue(Files.exists(modelFile));
+
+        final ApplicationModel copy = BootstrapUtils.readAppModelWithWorkspaceId(modelFile, workspaceId);
+
+        assertSameApplicationArtifact(model, copy);
+    }
+
+    private static ApplicationModel newApplicationModel() {
+        return new ApplicationModelBuilder()
+                .setAppArtifact(ResolvedDependencyBuilder.newInstance()
+                        .setCoords(ArtifactCoords.fromString(APP_COORDINATES))
+                        .setResolvedPaths(PathList.empty()))
+                .build();
+    }
+
+    private static void assertSameApplicationArtifact(final ApplicationModel expected, final ApplicationModel actual) {
+        assertNotNull(actual);
+        assertEquals(expected.getAppArtifact().toGACTVString(), actual.getAppArtifact().toGACTVString());
+        assertEquals(0, actual.getDependencies().size());
+    }
+}

--- a/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/src/test/java/io_quarkus/quarkus_bootstrap_app_model/DirectoryPathTreeTest.java
+++ b/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/src/test/java/io_quarkus/quarkus_bootstrap_app_model/DirectoryPathTreeTest.java
@@ -1,0 +1,75 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package io_quarkus.quarkus_bootstrap_app_model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import io.quarkus.paths.DirectoryPathTree;
+import io.quarkus.paths.PathFilter;
+
+public class DirectoryPathTreeTest {
+    @TempDir
+    Path tempDir;
+
+    @Test
+    public void serializesAndDeserializesDirectoryTreeState() throws Exception {
+        final Path root = tempDir.toAbsolutePath();
+        final Path visibleFile = root.resolve("visible.txt");
+        final Path hiddenFile = root.resolve("hidden.txt");
+        Files.writeString(visibleFile, "visible");
+        Files.writeString(hiddenFile, "hidden");
+        final DirectoryPathTree tree = new DirectoryPathTree(
+                root,
+                PathFilter.forIncludes(List.of("visible.txt")),
+                true);
+
+        final DirectoryPathTree copy = serializeAndDeserialize(tree);
+
+        assertEquals(tree, copy);
+        assertEquals(tree.hashCode(), copy.hashCode());
+        assertFalse(copy.isArchiveOrigin());
+        assertTrue(copy.isOpen());
+        assertFalse(copy.isEmpty());
+        assertEquals(List.of(root), List.copyOf(copy.getRoots()));
+        assertTrue(copy.contains("visible.txt"));
+        assertEquals(visibleFile, copy.getPath("visible.txt"));
+        assertFalse(copy.contains("hidden.txt"));
+        assertNull(copy.getPath("hidden.txt"));
+        assertEquals(copy, copy.getOriginalTree());
+    }
+
+    private static DirectoryPathTree serializeAndDeserialize(final DirectoryPathTree tree)
+            throws IOException, ClassNotFoundException {
+        final byte[] serialized;
+        try (ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+                ObjectOutputStream output = new ObjectOutputStream(bytes)) {
+            output.writeObject(tree);
+            output.flush();
+            serialized = bytes.toByteArray();
+        }
+
+        try (ObjectInputStream input = new ObjectInputStream(new ByteArrayInputStream(serialized))) {
+            return assertInstanceOf(DirectoryPathTree.class, input.readObject());
+        }
+    }
+}

--- a/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/src/test/java/io_quarkus/quarkus_bootstrap_app_model/DirectoryPathTreeTest.java
+++ b/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/src/test/java/io_quarkus/quarkus_bootstrap_app_model/DirectoryPathTreeTest.java
@@ -45,8 +45,7 @@ public class DirectoryPathTreeTest {
 
         final DirectoryPathTree copy = serializeAndDeserialize(tree);
 
-        assertEquals(tree, copy);
-        assertEquals(tree.hashCode(), copy.hashCode());
+        assertEquals(root.toString(), copy.toString());
         assertFalse(copy.isArchiveOrigin());
         assertTrue(copy.isOpen());
         assertFalse(copy.isEmpty());

--- a/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/src/test/java/io_quarkus/quarkus_bootstrap_app_model/LazySourceDirTest.java
+++ b/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/src/test/java/io_quarkus/quarkus_bootstrap_app_model/LazySourceDirTest.java
@@ -1,0 +1,75 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package io_quarkus.quarkus_bootstrap_app_model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import io.quarkus.bootstrap.workspace.LazySourceDir;
+import io.quarkus.paths.PathFilter;
+
+public class LazySourceDirTest {
+    @TempDir
+    Path tempDir;
+
+    @Test
+    public void serializesAndDeserializesConfiguredDirectoriesAndData() throws Exception {
+        final Path sourcePath = tempDir.resolve("src/main/java");
+        final Path outputPath = tempDir.resolve("target/classes");
+        final Path generatedSourcesPath = tempDir.resolve("target/generated-sources/annotations");
+        final PathFilter sourceFilter = PathFilter.forIncludes(List.of("**/*.java"));
+        final PathFilter outputFilter = PathFilter.forExcludes(List.of("**/generated/**"));
+        final Map<Object, Object> data = new HashMap<>();
+        data.put("language", "java");
+
+        final LazySourceDir sourceDir = new LazySourceDir(
+                sourcePath,
+                sourceFilter,
+                outputPath,
+                outputFilter,
+                generatedSourcesPath,
+                data);
+
+        final LazySourceDir copy = serializeAndDeserialize(sourceDir);
+
+        assertEquals(sourcePath.toAbsolutePath(), copy.getDir());
+        assertEquals(outputPath.toAbsolutePath(), copy.getOutputDir());
+        assertEquals(generatedSourcesPath.toAbsolutePath(), copy.getAptSourcesDir());
+        assertEquals("java", copy.getValue("language", String.class));
+        assertTrue(copy.toString().contains("src-filter="));
+        assertTrue(copy.toString().contains("dest-filter="));
+    }
+
+    private LazySourceDir serializeAndDeserialize(final LazySourceDir sourceDir)
+            throws IOException, ClassNotFoundException {
+        final byte[] serialized;
+        try (ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+                ObjectOutputStream output = new ObjectOutputStream(bytes)) {
+            output.writeObject(sourceDir);
+            output.flush();
+            serialized = bytes.toByteArray();
+        }
+
+        try (ObjectInputStream input = new ObjectInputStream(new ByteArrayInputStream(serialized))) {
+            return assertInstanceOf(LazySourceDir.class, input.readObject());
+        }
+    }
+}

--- a/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/src/test/java/io_quarkus/quarkus_bootstrap_app_model/PathTreeWithManifestTest.java
+++ b/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/src/test/java/io_quarkus/quarkus_bootstrap_app_model/PathTreeWithManifestTest.java
@@ -1,0 +1,25 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package io_quarkus.quarkus_bootstrap_app_model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.paths.PathTreeWithManifest;
+
+public class PathTreeWithManifestTest {
+    @Test
+    public void exposesCurrentJavaVersion() {
+        final int expectedJavaVersion = Runtime.version().feature();
+        final int actualJavaVersion = PathTreeWithManifest.JAVA_VERSION;
+
+        assertEquals(expectedJavaVersion, actualJavaVersion);
+        assertTrue(actualJavaVersion > 0);
+    }
+}

--- a/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/user-code-filter.json
+++ b/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/user-code-filter.json
@@ -1,0 +1,10 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "io.quarkus.**"
+    }
+  ]
+}

--- a/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/user-code-filter.json
+++ b/tests/src/io.quarkus/quarkus-bootstrap-app-model/3.34.0/user-code-filter.json
@@ -5,6 +5,9 @@
     },
     {
       "includeClasses" : "io.quarkus.**"
+    },
+    {
+      "includeClasses" : "io_quarkus.quarkus_bootstrap_app_model.**"
     }
   ]
 }

